src/Modules/OrchardCore.Commerce/Liquid/ToAmountConverterFilter.cs

```diff
@@ -0,0 +1,19 @@
+using Fluid;
+using Fluid.Values;
+using Newtonsoft.Json.Linq;
+using OrchardCore.Commerce.MoneyDataType;
+using OrchardCore.Liquid;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Liquid;
+
+public class ToAmountConverterFilter : ILiquidFilter
+{
+    public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+    {
+        if (input.Type == FluidValues.Nil) return new ValueTask<FluidValue>(input);
+
+        var newObject = (input.ToObjectValue() as JToken).ToObject<Amount>();
```

Comment:
`as` can return `null` if the type doesn't match, so this filter will throw NRE if used on the wrong input.